Барнаева М. К. Лаб. 3 Группа 6311

Barnaeva opened the pull request:
**ФИО:** Барнаева Марина  
**Номер группы:** 6311
**Номер лабораторной:** 3  **Номер варианта(при наличии):** 3
**Краткое описание проделанной работы:** Я выявила невалидные строки из файла csv при помощи регулярных выражений.

VornadRkk commented on lab_2/baseFunctions.py:
```diff
@@ -0,0 +1,41 @@
+import librosa
+import os
+from scipy.signal import spectrogram
+from display import read_file
```
Для лучшей читабельности кода рекомендую импортировать модули в алфавитном порядке и разделять пустой строкой разные группы импортов.

VornadRkk replied:
Для лучшей читабельности кода рекомендую импортировать модули в алфавитном порядке и разделять пустой строкой разные группы импортов.

VornadRkk commented on lab_2/baseFunctions.py:
```diff
@@ -0,0 +1,41 @@
+def get_data_set(dyrectory):
+
+    files_name = os.listdir(dyrectory)
+    y = []
+    files_names =[]
+    srs = []
```
Несоответствие стандартам кодирования PEP8( название переменных )
Рекомендую использовать более подробное описание переменных, также придерживаться стандартов именования переменных.

VornadRkk commented on lab_2/baseFunctions.py:
```diff
@@ -0,0 +1,41 @@
+
+    files_name = os.listdir(dyrectory)
+    y = []
+    files_names =[]
```
Несоответствие стандартам кодирования PEP8( пробел между оператором и операндом )
Принято ставить пробел между оператором и операндом ,для более удобного чтения кода.

VornadRkk commented on lab_2/baseFunctions.py:
```diff
@@ -0,0 +1,41 @@
+
+def define_class(y1, y2):
+    dec = 0
+    if (("kitchen" in y1) and ("kitchen" in y2)) and (("mi" in y1) and ("mi" in y2)): dec = 1
+    if (("kitchen" in y1) and ("kitchen" in y2)) and (("android" in y1) and ("android" in y2)): dec = 1
+    if (("kitchen" in y1) and ("kitchen" in y2)) and (("iphone" in y1) and ("iphone" in y2)): dec = 1
+    if (("room" in y1) and ("room" in y2)) and (("mi" in y1) and ("mi" in y2)): dec = 1
+    if (("room" in y1) and ("room" in y2)) and (("android" in y1) and ("android" in y2)): dec = 1
+    if (("room" in y1) and ("room" in y2)) and (("iphone" in y1) and ("iphone" in y2)): dec = 1
```
Несоответствие стандартам кодирования PEP8( длина строк )
Строки не должны быть длиннее 79 символов. Если строка слишком длинная, ее рекомендуется разбить на несколько строк, используя скобки, запятые или операторы конкатенации. Данная проблема встречается регулярно, для удобного чтения кода рекомендую обратить на это внимание.

VornadRkk commented on lab_2/statisticsClassification.py:
```diff
@@ -0,0 +1,95 @@
+all_vec = chi_points.reshape(-1,1)
+
+
+chi_x_train, chi_x_test, chi_y_train, chi_y_test = train_test_split(chi_points, labels, test_size=0.30)
+umann_x_train, umann_x_test, umann_y_train, umann_y_test = train_test_split(umann_points, labels, test_size=0.30)
+stud_x_train, stud_x_test, stud_y_train, stud_y_test = train_test_split(student_points, labels, test_size=0.30)
+mid_x_train, mid_x_test, mid_y_train, mid_y_test = train_test_split(middles, labels, test_size=0.30)
+all_x_train, all_x_test, all_y_train, all_y_test = train_test_split(all_vec, labels, test_size=0.30)
```
Несоответствие стандартам кодирования PEP8( длина строк )
Строки не должны быть длиннее 79 символов. Если строка слишком длинная, ее рекомендуется разбить на несколько строк, используя скобки, запятые или операторы конкатенации.

VornadRkk commented on lab_2/statisticsClassification.py:
```diff
@@ -0,0 +1,95 @@
+middles_pred = mid_model.predict(mid_x_test)
+all_pred = all_model.predict(all_x_test)
+
+report_chi = [accuracy_score(chi_y_test, chi_pred), f1_score(chi_y_test, chi_pred), precision_score(chi_y_test, chi_pred), recall_score(chi_y_test, chi_pred)]
+report_uman = [accuracy_score(umann_y_test, umann_pred), f1_score(umann_y_test, umann_pred), precision_score(umann_y_test, umann_pred), recall_score(umann_y_test, umann_pred)]
+report_stud = [accuracy_score(stud_y_test, stud_pred), f1_score(stud_y_test, stud_pred), precision_score(stud_y_test, stud_pred), recall_score(stud_y_test, stud_pred)]
+report_mid = [accuracy_score(mid_y_test, middles_pred), f1_score(mid_y_test, middles_pred), precision_score(mid_y_test, middles_pred), recall_score(mid_y_test, middles_pred)]
+report_all = [accuracy_score(all_y_test, all_pred), f1_score(all_y_test, all_pred), precision_score(all_y_test, all_pred), recall_score(all_y_test, all_pred)]
```
Несоответствие стандартам кодирования PEP8( длина строк )
Строки не должны быть длиннее 79 символов. Если строка слишком длинная, ее рекомендуется разбить на несколько строк, используя скобки, запятые или операторы конкатенации. Данная проблема встречается регулярно, для удобного чтения кода рекомендую обратить на это внимание.

VornadRkk commented on lab_2/statisticsClassification.py:
```diff
@@ -0,0 +1,95 @@
+        all_vec.append(vec)
+
+df = pandas.DataFrame({'chi': chi_points, 'unamm': umann_points, 'stud': student_points, 'pair': labels, 'middle': middles})
+df.to_csv('stats.csv')
```
Проблема с безопасностью
Лучше не хардкодить имя файла базы данных. Рекомендуется хранить путь файл в отдельном файле и передавать как параметр функции.

VornadRkk commented on lab_2/baseFunctions.py:
```diff
@@ -0,0 +1,41 @@
+    if (("second_pink" in y1) and ("second_pink" in y2)): dec = 1
+
+    return dec
+
+def spectrogramm(x):
```
Несоответствие стандартам кодирования PEP8( функции)
Рекомендуется ставить 2 пробела между функциями для лучшей читабельности кода.

VornadRkk reviewed (commented):
Несоответствие стандартам кодирования PEP8

Не используется глобальная функция main()

VornadRkk reviewed (commented):
Несоответствие стандартам кодирования PEP8

В обоих файлах отсутствует пробел в конце файла

VornadRkk reviewed (commented):
Несоответствие стандартам кодирования PEP8

В обоих файлах отсутствует type hint и docstring. Рекомендуется их использовать ,так как это делает код более понятным.

VornadRkk reviewed (commented):
Несоответствие стандартам кодирования PEP8

В обоих файлах множество лишних пробелов, рекомендую от них избавиться, так как уменьшиться количество строк кода.

IlyaOv commented:
Подробно разобрано на несоответствие стандартам кодирования, хорошо. Но хотелось бы увидеть варианты улучшения архитектуры кода (что делать с дублирование, например).

VornadRkk reviewed (commented):
Рекомендую вынести повторяющиеся участки кода в отдельные функции чтобы в дальнейшем вызывать эти функции, для того чтобы выполнялся принцип DRY

IlyaOv commented:
Да, но хотелось бы увидеть указание на конкретный участок и пример функции.

VornadRkk commented on lab_2/statisticsClassification.py:
```diff
@@ -0,0 +1,95 @@
+mid_x_train, mid_x_test, mid_y_train, mid_y_test = train_test_split(middles, labels, test_size=0.30)
+all_x_train, all_x_test, all_y_train, all_y_test = train_test_split(all_vec, labels, test_size=0.30)
+
+chi_model.fit(chi_x_train, chi_y_train)
+umann_model.fit(umann_x_train, umann_y_train)
+stud_model.fit(stud_x_train, stud_y_train)
+mid_model.fit(mid_x_train, mid_y_train)
+all_model.fit(X=all_x_train, y=all_y_train)
```
Нарушен принцип DRY ,предлагаю сделать функцию и вызывать ее:
def train_and_predict(model, x_train, y_train, x_test):
    model.fit(x_train, y_train)
    return model.predict(x_test)
   С помощью этой функции мы сможем сократить кол-во строк в коде, объединив 72-76 строки и 79-83
   в одну функцию.
   Также принцип DRY нарушен еще в некоторых участках кода, советую проверить их и исправить

IlyaOv commented:
Зачтено

Barnaeva committed:
test

Barnaeva committed:
Fixed import order to comply with PEP 8 standards

Barnaeva committed:
Fixed E501 (too long line) according to PEP 8 recommendations.

Barnaeva committed:
F401: Unused imports removed
E302: Added blank lines between functions
E251: Fixed spacing around parameters

Barnaeva committed:
Fixed E128, E129, E131, E722 in accordance with PEP 8

Barnaeva committed:
Fixed F841 in accordance with PEP 8

Barnaeva committed:
Fixed N806,E123 in accordance with PEP 8

Barnaeva committed:
N802, N803, N806 - Variable/function naming
E401, E402 - Import order and grouping

Barnaeva committed:
a file has been added for working with json and csv

Barnaeva committed:
+patterns,+settings,fixing the io_operations file

Barnaeva committed:
serialize_result function, module for working with validity definitions, main, editing settings files, patterns

Barnaeva committed:
PEP8

IlyaOv reviewed (commented):
Поправьте импорты в соответствии с [PEP8](https://peps.python.org/pep-0008/#imports)

Barnaeva committed:
Update pattern_fun.py

Barnaeva committed:
pep8

Barnaeva committed:
pep8

Barnaeva commented:
Исправлено.

IlyaOv reviewed (approved):
Зачтено
